storage/innobase/fil/fil0crypt.cc

```diff
@@ -2364,6 +2364,24 @@ fil_crypt_set_rotation_iops(
 	os_event_set(fil_crypt_threads_event);
 }
 
+/** Add the import tablespace to default_encrypt list
+if necessary and signal fil_crypt_threads
+@param space imported tablespace */
+void fil_crypt_add_imported_space(fil_space_t *space)
+{
+  mutex_enter(&fil_system.mutex);
+
+  if (fil_crypt_must_default_encrypt())
```

Comment:
It would be good to assert in `fil_crypt_must_default_encrypt()` that `fil_system.mutex` is being held, with a comment saying that it prevents a race condition with `fil_crypt_set_rotate_key_age()`.